File: repository/extend-maven/src/main/java/org/abstracthorizon/extend/repo/maven/MavenArtifact.java
package org.abstracthorizon.extend.repo.maven;

import org.abstracthorizon.extend.repo.Artifact;
import org.abstracthorizon.extend.repo.ArtifactImpl;

public class MavenArtifact {

    public static Artifact apply(String s) {
        String[] list = s.split(":");
        if (list.length == 2) {
            return ArtifactImpl.apply(list[0], list[1]);
        } else if (list.length == 3) {
            return ArtifactImpl.apply(list[0], list[1], MavenVersion.apply(list[2]));
        } else if (list.length == 4) {
            return ArtifactImpl.apply(list[0], list[1], MavenVersion.apply(list[2]), list[3]);
        } else if (list.length == 5) {
            return ArtifactImpl.apply(list[0], list[1], MavenVersion.apply(list[2]), list[3], list[4]);
        } else {
            return null;
        }
    }
 
    public static Artifact toPOM(Artifact artifact) {
        if ("pom".equals(artifact.getType())) {
            return artifact;
        } else {
            return ArtifactImpl.apply(artifact.getGroupId(), artifact.getArtifactId(), artifact.getVersion(), "pom", artifact.getClassifier());
        }
    }
    
    public static Artifact toFinal(Artifact artifact) {
        if (artifact.getVersion().isFinal()) {
            return artifact;
        } else { 
            return ArtifactImpl.apply(artifact.getGroupId(), artifact.getArtifactId(), artifact.getVersion().toFinal(), artifact.getType(), artifact.getClassifier());
        }
    }
    
    public static Artifact toSnapshot(Artifact artifact) {
        if (!artifact.getVersion().isFinal()) {
            return artifact;
        } else { 
            return ArtifactImpl.apply(artifact.getGroupId(), artifact.getArtifactId(), MavenVersion.toSnapshot(artifact.getVersion()), artifact.getType(), artifact.getClassifier());
        }
        
    }
}

/*


    def toSnapshot(artifact: Artifact): Artifact = if (!artifact.version.isFinal) artifact else Artifact(artifact.groupId, artifact.artifactId, MavenVersion.toSnapshot(artifact.version), artifact.typ, artifact.classifier)
*/

File: repository/extend-scader/src/main/scala/org/abstracthorizon/extend/repository/maven/MavenArtifact.scala
package org.abstracthorizon.extend.repository.maven

import org.abstracthorizon.extend.repository._

object MavenArtifact {
    def apply(s: String): Artifact = {
        val list = s.split(":")
        if (list.length == 2) {
            return Artifact.apply(list(0), list(1))
        } else if (list.length == 3) {
            return Artifact.apply(list(0), list(1), MavenVersion(list(2)))
        } else if (list.length == 4) {
            return Artifact.apply(list(0), list(1), MavenVersion(list(2)), list(3))
        } else if (list.length == 5) {
            return Artifact.apply(list(0), list(1), MavenVersion(list(2)), list(3), list(4))
        } else {
            return null
        }
    }

    def toPOM(artifact: Artifact): Artifact = if ("pom".equals(artifact.typ)) artifact else Artifact(artifact.groupId, artifact.artifactId, artifact.version, "pom", artifact.classifier)

    def toFinal(artifact: Artifact): Artifact = if (artifact.version.isFinal) artifact else Artifact(artifact.groupId, artifact.artifactId, artifact.version.toFinal, artifact.typ, artifact.classifier)

    def toSnapshot(artifact: Artifact): Artifact = if (!artifact.version.isFinal) artifact else Artifact(artifact.groupId, artifact.artifactId, MavenVersion.toSnapshot(artifact.version), artifact.typ, artifact.classifier)
}